compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -724,4 +712,52 @@ object TreeChecker {
 
     override def simplify(tree: Tree, pt: Type, locked: TypeVars)(using Context): tree.type = tree
   }
+
+  /** Tree checker that can be applied to a local tree. */
+  class LocalChecker(phasesToCheck: Seq[Phase]) extends Checker(phasesToCheck: Seq[Phase]):
+    override def assertDefined(tree: untpd.Tree)(using Context): Unit =
+      // Only check definitions nested in the local tree
+      if nowDefinedSyms.contains(tree.symbol.maybeOwner) then
+        super.assertDefined(tree)
+
+  def checkMacroGeneratedTree(original: tpd.Tree, expansion: tpd.Tree)(using Context): Unit =
+    if ctx.settings.XcheckMacros.value then
+      val checkingCtx = ctx
+        .fresh
+        .addMode(Mode.ImplicitsEnabled)
```

Comment:
Do we really need to generate implicits when checking trees?

Reply:
I followed the original tree checker https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/transform/TreeChecker.scala#L137.

I will try removing both.